tests/src/providers/testqgspostgresconn.cpp

```diff
@@ -38,6 +40,35 @@ class TestQgsPostgresConn: public QObject
       QCOMPARE( QgsPostgresConn::quotedValue( "b \"c' \\x" ), QString( "E'b \"c'' \\\\x'" ) );
     }
 
+    void quotedValueDatetime()
+    {
+      QCOMPARE( QgsPostgresConn::quotedValue( QDateTime::fromString( "2020-05-07 17:56:00", "yyyy-MM-dd HH:mm:ss" ) ), QString( "'2020-05-07T17:56:00.000'" ) );
+
+      QgsFields fields;
+      QgsField f;
+      f.setName( "ts" );
+      f.setType( QVariant::DateTime );
+      f.setTypeName( "timestamp" );
+      fields.append( f );
+      QgsField f2;
+      f2.setName( "pk" );
+      f2.setType( QVariant::ULongLong );
```

Comment:
That might be the reason your test is failing (2020-06-07 22:20:00-03:00): when telling a field type is one not recognized by QGIS, QgsPostgresConn::fieldExpression() returns a cast to text. You should just set your type to QVariant::LongLong here, and it'll be good. Thanks for your work!

Reply:
argg... so far so good

Reply:
done!

Reply:
Hi, @benoitdm-oslandia , I've cloned your repo and ran the tests locally, with the command
`ctest -V -R qgis_postgresconntest`

 I've found that if you change ULL to LL on line 66 here: https://github.com/qgis/QGIS/blob/f1bb526df8bbf5a304d5458085b7a0264235dab8/tests/src/providers/testqgspostgresconn.cpp#L63-L67 the test pass on my system.

If I may ask, what does QVariant::LastValue mean here? I couldn't find any docs about it, only the definition in /usr/include/qt5/QtCore/qvariant.h:

```c++
LastType = 0xffffffff // need this so that gcc >= 3.4 allocates 32 bits for Type
```

Is it thread safe? Thanks!

Reply:
sorry I was so lazy, I did not run the test to check!